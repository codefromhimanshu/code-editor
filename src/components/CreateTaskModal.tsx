
import React, { useState } from 'react';

interface CreateTaskModalProps {
  isOpen: boolean;
  onClose: () => void;
  onSubmit: (title: string, description: string) => void;
}

const CreateTaskModal: React.FC<CreateTaskModalProps> = ({ isOpen, onClose, onSubmit }) => {
  const [title, setTitle] = useState('');
  const [description, setDescription] = useState('');

  const handleSubmit = (e: React.FormEvent) => {
    e.preventDefault();
    onSubmit(title, description);
    setTitle('');
    setDescription('');
  };

  if (!isOpen) return null;

  return (
    <div className="fixed inset-0 z-50 overflow-auto bg-smoke-light flex">
      <div className="relative p-8 bg-white w-full max-w-md m-auto flex-col flex rounded-lg">
        <span className="absolute top-0 right-0 p-4">
          <button onClick={onClose}>&times;</button>
        </span>
        <form onSubmit={handleSubmit}>
          <div className="mb-5">
            <label htmlFor="title" className="font-semibold mb-2 block">Task Title</label>
            <input
              id="title"
              type="text"
              value={title}
              onChange={(e) => setTitle(e.target.value)}
              className="border py-2 px-3 text-grey-darkest w-full"
              required
            />
          </div>
          <div className="mb-5">
            <label htmlFor="description" className="font-semibold mb-2 block">Description</label>
            <textarea
              id="description"
              value={description}
              onChange={(e) => setDescription(e.target.value)}
              className="border py-2 px-3 text-grey-darkest w-full"
              required
            />
          </div>
          <div className="flex justify-end">
            <button
              type="submit"
              className="bg-blue-500 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded"
            >
              Create Task
            </button>
          </div>
        </form>
      </div>
    </div>
  );
};

export default CreateTaskModal;
