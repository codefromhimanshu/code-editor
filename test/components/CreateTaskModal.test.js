// tests/CreateTaskModal.test.js
import React from "react";
import renderer from "react-test-renderer";
import CreateTaskModal from "../../src/components/CreateTaskModal";

describe("CreateTaskModal", () => {
  it("renders correctly when open", () => {
    const mockOnSubmit = jest.fn();
    const mockOnClose = jest.fn();
    const tree = renderer
      .create(
        <CreateTaskModal
          isOpen={true}
          onClose={mockOnClose}
          onSubmit={mockOnSubmit}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("does not render when not open", () => {
    const mockOnSubmit = jest.fn();
    const mockOnClose = jest.fn();
    const tree = renderer
      .create(
        <CreateTaskModal
          isOpen={false}
          onClose={mockOnClose}
          onSubmit={mockOnSubmit}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
