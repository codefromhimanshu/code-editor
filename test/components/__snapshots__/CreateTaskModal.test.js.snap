// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateTaskModal does not render when not open 1`] = `null`;

exports[`CreateTaskModal renders correctly when open 1`] = `
<div
  className="fixed inset-0 z-50 overflow-auto bg-smoke-light flex"
>
  <div
    className="relative p-8 bg-white w-full max-w-md m-auto flex-col flex rounded-lg"
  >
    <span
      className="absolute top-0 right-0 p-4"
    >
      <button
        onClick={[MockFunction]}
      >
        Ã—
      </button>
    </span>
    <form
      onSubmit={[Function]}
    >
      <div
        className="mb-5"
      >
        <label
          className="font-semibold mb-2 block"
          htmlFor="title"
        >
          Task Title
        </label>
        <input
          className="border py-2 px-3 text-grey-darkest w-full"
          id="title"
          onChange={[Function]}
          required={true}
          type="text"
          value=""
        />
      </div>
      <div
        className="mb-5"
      >
        <label
          className="font-semibold mb-2 block"
          htmlFor="description"
        >
          Description
        </label>
        <textarea
          className="border py-2 px-3 text-grey-darkest w-full"
          id="description"
          onChange={[Function]}
          required={true}
          value=""
        />
      </div>
      <div
        className="flex justify-end"
      >
        <button
          className="bg-blue-500 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded"
          type="submit"
        >
          Create Task
        </button>
      </div>
    </form>
  </div>
</div>
`;
